### PR TITLE
🔥  Fix pubmine ads

### LINKED ISSUE #22429: Pubmine ads don't work 1905211840370
## What's the issue?

Pubmine ads don't work.

## How do we reproduce the issue?

Open a page containing a pubmine ad.

The problem can't be reproduced using instructions described [here](https://github.com/ampproject/amphtml/blob/master/ads/README.md#verify-your-examples). Therefore it wasn't noticed during QA and code review.

## What browsers are affected?

All browsers

## Which AMP version is affected?

1905211840370
1905140117570/production

#22426

### PR DESCRIPTION
**Pull request by shackijj**
Fixes #22429

### CODE REVIEW

**Comment by prateekbh** [comment]
Also can you open a cherry-pick request to canary for this

File: ads/pubmine.js [hunk 1]
```diff
@@ -22,8 +22,10 @@ const pubmineOptional = ['section', 'pt', 'ht'],
 
 /**
  * @param {!Object} data
+ * @param {!Window} global
+ * @param {!Function} [scriptLoader=loadScript]
  */
-function initMasterFrame(data) {
+function initMasterFrame(data, global, scriptLoader = loadScript) {
```
Comment (calebcordry) [inline ads/pubmine.js]: I understand why global may not behave how we would expect here, but I am curious why `loadScript` needs to be passed around. Are you just being extra cautious here?

File: test/unit/ads/test-pubmine.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
```
Comment (calebcordry) [inline test/unit/ads/test-pubmine.js]: nit: 2019

File: ads/pubmine.js (hunk 1 above)
Comment (shackijj) [reply ads/pubmine.js]: It's an easy way to pass a spy. I check if `loadScript` was called in unit tests.

Comment (calebcordry) [reply ads/pubmine.js]: Got it. Thanks for clarifying.

Comment (lannka) [reply ads/pubmine.js]: agree, `scriptLoader ` is not needed here

Comment (shackijj) [reply ads/pubmine.js]: Let's decide what we should do here. In my opinion, it's fine to pass the function. If stubbing `loadScript` in tests is more preferable I can implement it.

**Comment by calebcordry** [comment]
@shackijj -- @lannka and I were chatting offline. What do you think about changing the tests for `loadScript` to then queryselect that the element is created? You could do the same for the `createSlot` result. Then we could avoid adding the params just for testing.

File: ads/pubmine.js (hunk 1 above)
Comment (calebcordry) [reply ads/pubmine.js]: Did you see the comment here https://github.com/ampproject/amphtml/pull/22431#issuecomment-494877854

File: test/unit/ads/test-pubmine.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        pt: 2,
+        ht: 2,
+        tn: 'amp',
+        amp: true,
+      };
+      pubmine(win, mockData);
+      expect(win.__ATA_PP).to.deep.equal(expectedConfig);
+      expect(win.__ATA.cmd).to.be.an('array');
+      expect(win.__ATA.cmd).to.have.length(1);
+      expect(getPubmineScriptElement()).not.to.equal(null);
```
Comment (lannka) [inline test/unit/ads/test-pubmine.js]: nit: `expect(..).to.be.ok`

File: test/unit/ads/test-pubmine.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      };
+      const mockData = {
+        siteid: 'amp-test',
+        section: 1,
+        pt: 2,
+        ht: 2,
+      };
+      pubmine(win, mockData);
+      expect(win.context.master.__ATA.cmd).to.have.length(1);
+      expect(getPubmineScriptElement()).to.equal(null);
```
Comment (lannka) [inline test/unit/ads/test-pubmine.js]: expect().to.be.null

File: test/unit/ads/test-pubmine.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    });
+  });
```
Comment (lannka) [inline test/unit/ads/test-pubmine.js]: to be a complete test, do you also want to check your slot element is created?

**Pull request merged**